benchmarks/pca/solvers/cuml.py

```diff
@@ -0,0 +1,89 @@
+from importlib.metadata import version
+
+from benchopt import BaseSolver, safe_import_context
+from benchopt.stopping_criterion import SingleRunCriterion
+
+with safe_import_context() as import_ctx:
+    import cuml
+    import cupy
+    import numpy as np
+
+
+class Solver(BaseSolver):
+    name = "cuml"
+    requirements = ["cuml"]
+
+    parameters = dict(
+        device=["gpu"],
+        svd_solver=["full", "jacobi"],
+    )
+
+    stopping_criterion = SingleRunCriterion(1)
+
+    def skip(self, **objective_dict):
+
+        X = objective_dict["X"]
+        if X.dtype == np.float64:
+            # We haven't came accross cuda devices that doesn't support float64 yet,
+            # can it happen ? If it happens, the following instruction will fail,
+            # please enclose it with the appropriate Try/Except to return the
+            # appropriate skip decision.
+            cupy.zeros(1, dtype=cupy.float64)
+            # return True, (
+            #     f"This {self.device} device has no support for float64 compute"
+            # )
+
+        return False, None
+
+    def set_objective(
+        self,
+        X,
+        n_components,
+        whiten,
+        tol,
+        iterated_power,
+        n_oversamples,
+        random_state,
+        verbose,
+    ):
+        if self.device == "cpu":
```

Comment:
I don't think `cuml.decomposition.PCA` can ever run on CPU. It can accept host-allocated inputs but it will do the device-allocation + copy automatically in that case.

In my opinion, let's not waste benchmark time and reporting readability to measure this: for the cuml case, I would only run the `device == "gpu"` case and remove the fake `device == "cpu"` case.

Reply:
(I don't really now the general whereabouts but cuml is introducing experimental [cpu / gpu device selection](https://docs.rapids.ai/api/cuml/stable/api/#device-selection) and [PCA](https://docs.rapids.ai/api/cuml/stable/api/#principal-component-analysis) is compatible so maybe at some point it could be included.)